docs/_ext/autoclass_with_examples.py

```diff
@@ -0,0 +1,94 @@
+import re
+import sys
+from typing import Any, Dict, List, Tuple
+
+from docutils.nodes import Node
+from sphinx.ext.autodoc.directive import AutodocDirective
+
+from starknet_py.constants import ROOT_PATH
+
+
+class AutoclassWithExamples(AutodocDirective):
+    """
+    Directive class extending AutodocDirective and adding code examples.
```

Comment:
Let's maybe write a short description of how the class works (like the fact that it adds raw `codesnippet` and uses code examples defined in ...../code_examples)